runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -663,6 +663,14 @@ gcParseXlpOption(J9JavaVM *vm)
 			j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_XLP_EXTRA_COMMA);
 		}
 	}
+	else if (vm->vmRuntimeStateListener.largePageSizeInBytes != NULL) {
+		/* try to get memory value */
+		if (!scan_udata_memory_size_helper(vm, &(vm->vmRuntimeStateListener.largePageSizeInBytes), &requestedPageSize, "")) {
+			/* size is not formed properly */
+			xlpErrorState = XLP_PAGE_SIZE_INCORRECT;
+		}
+		xlpErrorState = XLP_NO_ERROR;
+	}
```

Comment:
This code in not accurate:
- `scan_udata_memory_size_helper` supports "memory" formats (4096, 4k, 128M, 2g). The `-XX:LargePageSizeInBytes=<size>` uses just decimal number. This unnecessary extension creating incompatibility. So do we want `-XX:LargePageSizeInBytes=16m` to work?
- Original parser returns `largePageFailOnError` and `largePageWarnOnError` conditions based on sizes available in Port Library (and OS). For example if 16m page requested but not provisioned in OS we need at least provide warning or even stop JVM execution. This code does not provide such coverage

Reply:
Another lost point is running on ZOS required `pageable` or `nonpageable` mandatory attribute. This case is not covered